src/Icicle/Avalanche/Check.hs

```diff
@@ -88,7 +88,13 @@ checkStatement frag ctx stmt
                    $ checkExp frag (ctxExp ctx) x
                Let n x' <$> go stmts
 
-        ForeachInts n from to stmts
+        While t n to stmts
+         -> do to'   <- first ProgramErrorExp
+                      $ checkExp frag (ctxExp ctx) to
+
+               While t n to' <$> go stmts
```

Comment:
yeah, I forgot about this. This should have a requireSame here, and the insert n below no longer applies.